### ISSUE TITLE
Gulp dist build fails to `uglify` in Node v6.0.0

### ISSUE THREAD

**Issue by tony19** [PolymerElements/polymer-starter-kit#808]
### Description

The `gulp` build (`default` or `serve:dist`) fails to complete `uglify` when running under Node v6.0.0, but has no issue in earlier Node versions. I added `$.uglify(...).on('error', gutil.log)`, which shows that it was trying to `uglify` a CSS file:
```
{ Error
    at new JS_Parse_Error (eval at <anonymous> (/Users/tony/Downloads/polymer-starter-kit-1.3.0/node_modules/uglify-js/tools/node.js:22:1), <anonymous>:1526:18)
    at js_error (eval at <anonymous> (/Users/tony/Downloads/polymer-starter-kit-1.3.0/node_modules/uglify-js/tools/node.js:22:1), <anonymous>:1534:11)
    at croak (eval at <anonymous> (/Users/tony/Downloads/polymer-starter-kit-1.3.0/node_modules/uglify-js/tools/node.js:22:1), <anonymous>:2026:9)
    at token_error (eval at <anonymous> (/Users/tony/Downloads/polymer-starter-kit-1.3.0/node_modules/uglify-js/tools/node.js:22:1), <anonymous>:2034:9)
    at unexpected (eval at <anonymous> (/Users/tony/Downloads/polymer-starter-kit-1.3.0/node_modules/uglify-js/tools/node.js:22:1), <anonymous>:2040:9)
    at semicolon (eval at <anonymous> (/Users/tony/Downloads/polymer-starter-kit-1.3.0/node_modules/uglify-js/tools/node.js:22:1), <anonymous>:2060:56)
    at simple_statement (eval at <anonymous> (/Users/tony/Downloads/polymer-starter-kit-1.3.0/node_modules/uglify-js/tools/node.js:22:1), <anonymous>:2240:73)
    at eval (eval at <anonymous> (/Users/tony/Downloads/polymer-starter-kit-1.3.0/node_modules/uglify-js/tools/node.js:22:1), <anonymous>:2113:19)
    at eval (eval at <anonymous> (/Users/tony/Downloads/polymer-starter-kit-1.3.0/node_modules/uglify-js/tools/node.js:22:1), <anonymous>:2073:24)
    at eval (eval at <anonymous> (/Users/tony/Downloads/polymer-starter-kit-1.3.0/node_modules/uglify-js/tools/node.js:22:1), <anonymous>:2827:23)
  message: '/Users/tony/Downloads/polymer-starter-kit-1.3.0/app/styles/main.css: Unexpected token: punc ({)',
  fileName: '/Users/tony/Downloads/polymer-starter-kit-1.3.0/app/styles/main.css',
  lineNumber: 10,
  stack: 'Error\n    at new JS_Parse_Error (eval at <anonymous> (/Users/tony/Downloads/polymer-starter-kit-1.3.0/node_modules/uglify-js/tools/node.js:22:1), <anonymous>:1526:18)\n    at js_error (eval at <anonymous> (/Users/tony/Downloads/polymer-starter-kit-1.3.0/node_modules/uglify-js/tools/node.js:22:1), <anonymous>:1534:11)\n    at croak (eval at <anonymous> (/Users/tony/Downloads/polymer-starter-kit-1.3.0/node_modules/uglify-js/tools/node.js:22:1), <anonymous>:2026:9)\n    at token_error (eval at <anonymous> (/Users/tony/Downloads/polymer-starter-kit-1.3.0/node_modules/uglify-js/tools/node.js:22:1), <anonymous>:2034:9)\n    at unexpected (eval at <anonymous> (/Users/tony/Downloads/polymer-starter-kit-1.3.0/node_modules/uglify-js/tools/node.js:22:1), <anonymous>:2040:9)\n    at semicolon (eval at <anonymous> (/Users/tony/Downloads/polymer-starter-kit-1.3.0/node_modules/uglify-js/tools/node.js:22:1), <anonymous>:2060:56)\n    at simple_statement (eval at <anonymous> (/Users/tony/Downloads/polymer-starter-kit-1.3.0/node_modules/uglify-js/tools/node.js:22:1), <anonymous>:2240:73)\n    at eval (eval at <anonymous> (/Users/tony/Downloads/polymer-starter-kit-1.3.0/node_modules/uglify-js/tools/node.js:22:1), <anonymous>:2113:19)\n    at eval (eval at <anonymous> (/Users/tony/Downloads/polymer-starter-kit-1.3.0/node_modules/uglify-js/tools/node.js:22:1), <anonymous>:2073:24)\n    at eval (eval at <anonymous> (/Users/tony/Downloads/polymer-starter-kit-1.3.0/node_modules/uglify-js/tools/node.js:22:1), <anonymous>:2827:23)',
  showStack: false,
  showProperties: true,
  plugin: 'gulp-uglify' }
```

The failure occurs in `optimizeHtmlTask()` when it tries to pass a filename pattern to `gulp-if`, which leads to a bug in `gulp-match` (https://github.com/robrich/gulp-match/issues/12), where string-conditions (i.e., `'*.js'`) aren't handled properly. A workaround is to use a function-condition instead:

**`gulpfile.js`:**
 1. Insert the following code:

 ```javascript
var filter = function(pattern) {
      return function(file) {
          if (typeof pattern === 'string') {
              var newCond = pattern.substring(1).replace(/\./g,'\\.')+'$';
              pattern = new RegExp(newCond);
          }
          return pattern.test(file.relative);
      };
};

var _if = function(pattern, trueChild, falseChild, minimatchOptions) {
        return $.if(filter(pattern), trueChild, falseChild, minimatchOptions);
};
```

 2. In `optimizeHtmlTask()`, replace `$.if` with `_if`.

*see StackOverflow [post](http://stackoverflow.com/questions/36901166/gulp-error-for-polymer-start-kit-on-osx)*

### Expected outcome
The build succeeds:
```
$ gulp
[19:47:39] Using gulpfile ~/Downloads/polymer-starter-kit-1.3.0/gulpfile.js
[19:47:39] Starting 'clean'...
[19:47:39] Finished 'clean' after 43 ms
[19:47:39] Starting 'default'...
[19:47:39] Starting 'ensureFiles'...
[19:47:39] Finished 'ensureFiles' after 338 μs
[19:47:39] Starting 'copy'...
[19:47:39] Starting 'styles'...
[19:47:39] styles all files 120 B
[19:47:39] Finished 'styles' after 358 ms
[19:47:39] copy all files 1.11 MB
[19:47:39] Finished 'copy' after 476 ms
[19:47:39] Starting 'images'...
[19:47:40] Starting 'fonts'...
[19:47:40] Starting 'html'...
[19:47:40] Finished 'fonts' after 136 ms
[19:47:41] html all files 54.8 kB
[19:47:41] Finished 'html' after 1.84 s
[19:47:42] gulp-imagemin: Minified 9 images (saved 1.27 kB - 1.9%)
[19:47:42] images all files 65.21 kB
[19:47:42] Finished 'images' after 2.6 s
[19:47:42] Starting 'vulcanize'...
[19:47:43] vulcanize all files 518.28 kB
[19:47:43] Finished 'vulcanize' after 1.44 s
[19:47:43] Finished 'default' after 4.51 s
```

### Actual outcome

```
$ gulp
(node:10720) fs: re-evaluating native module sources is not supported. If you are using the graceful-fs module, please update it to a more recent version.
[19:48:28] Using gulpfile ~/Downloads/polymer-starter-kit-1.3.0/gulpfile.js
[19:48:28] Starting 'clean'...
[19:48:28] Finished 'clean' after 38 ms
[19:48:28] Starting 'default'...
[19:48:28] Starting 'ensureFiles'...
[19:48:28] Finished 'ensureFiles' after 427 μs
[19:48:28] Starting 'copy'...
[19:48:28] Starting 'styles'...
[19:48:28] styles all files 120 B
[19:48:28] Finished 'styles' after 392 ms
[19:48:29] copy all files 1.11 MB
[19:48:29] Finished 'copy' after 529 ms
[19:48:29] Starting 'images'...
[19:48:29] Starting 'fonts'...
[19:48:29] Starting 'html'...
[19:48:29] Finished 'fonts' after 124 ms

events.js:160
      throw er; // Unhandled 'error' event
      ^
Error
    at new JS_Parse_Error (eval at <anonymous> (/Users/tony/Downloads/polymer-starter-kit-1.3.0/node_modules/uglify-js/tools/node.js:22:1), <anonymous>:1526:18)
    at js_error (eval at <anonymous> (/Users/tony/Downloads/polymer-starter-kit-1.3.0/node_modules/uglify-js/tools/node.js:22:1), <anonymous>:1534:11)
    at croak (eval at <anonymous> (/Users/tony/Downloads/polymer-starter-kit-1.3.0/node_modules/uglify-js/tools/node.js:22:1), <anonymous>:2026:9)
    at token_error (eval at <anonymous> (/Users/tony/Downloads/polymer-starter-kit-1.3.0/node_modules/uglify-js/tools/node.js:22:1), <anonymous>:2034:9)
    at unexpected (eval at <anonymous> (/Users/tony/Downloads/polymer-starter-kit-1.3.0/node_modules/uglify-js/tools/node.js:22:1), <anonymous>:2040:9)
    at semicolon (eval at <anonymous> (/Users/tony/Downloads/polymer-starter-kit-1.3.0/node_modules/uglify-js/tools/node.js:22:1), <anonymous>:2060:56)
    at simple_statement (
```

### Steps to reproduce
1. Download [polymer-starter-kit-1.3.0.zip](https://github.com/PolymerElements/polymer-starter-kit/releases/download/v1.3.0/polymer-starter-kit-1.3.0.zip).
2. Unzip, and open shell to the unzipped directory.
3. *Optional:* Install Node v6.0.0 with Homebrew.
4. Run `npm install && bower install`
5. Run `gulp` or `gulp serve:dist`.

### Environment
* **Operating System:** OS X El Capitan
* **Shell:** zsh
* **Node Version:** 6.0.0

**Comment by TimvdLippe** [comment]
What an awesome description, was a pleasure to read, good job! My 2 cents: wait for https://github.com/robrich/gulp-match/pull/13 to be merged and then verify it fixes the issue for the PSK. Temporarily switching to a custom `if` implementation might result in unwanted side-effects.

**Comment by tony19** [comment]
https://github.com/robrich/gulp-match/pull/13 was just merged and released in [gulp-match@1.0.2](https://github.com/robrich/gulp-match/releases/tag/1.0.2), which is now being pulled in by PSK on `npm install`. I verified the issue no longer occurs.
